plugins/ExtendedTools/objmgr.c

```diff
@@ -1158,64 +1300,65 @@ NTSTATUS EtEnumCurrentDirectoryObjects(
 
     PRESOLVER_THREAD_CONTEXT threadContext = PhAllocateZero(sizeof(RESOLVER_THREAD_CONTEXT));
     threadContext->Context = Context;
-    Context->BreakResolverThread = &threadContext->Break;
+    Context->BreakResolverThread = (PBOOLEAN)&threadContext->Break;
 
     PhCreateThread2(EtpTargetResolverThreadStart, threadContext);
 
     return STATUS_SUCCESS;
 }
 
 VOID EtObjectManagerFreeListViewItems(
-    _In_ POBJECT_CONTEXT Context
+    _In_ PET_OBJECT_CONTEXT Context
     )
 {
     INT index = INT_ERROR;
 
     if (Context->BreakResolverThread)
     {
-        *Context->BreakResolverThread = TRUE;
+        WriteRelease8(Context->BreakResolverThread, TRUE);
         Context->BreakResolverThread = NULL;
     }
 
-    PhClearReference(&Context->CurrentPath);
+    PhClearReference((PVOID)&Context->CurrentPath);
 
     while ((index = PhFindListViewItemByFlags(
         Context->ListViewHandle,
         index,
         LVNI_ALL
         )) != INT_ERROR)
     {
-        POBJECT_ENTRY entry;
+        PET_OBJECT_ENTRY entry;
 
         if (PhGetListViewItemParam(Context->ListViewHandle, index, &entry))
         {
-            PhClearReference(&entry->Name);
-            PhClearReference(&entry->TypeName);
-            if (entry->Target)
-                PhClearReference(&entry->Target);
-            if (entry->TargetDrvLow)
-                PhClearReference(&entry->TargetDrvLow);
-            if (entry->TargetDrvUp)
-                PhClearReference(&entry->TargetDrvUp);
-            PhFree(entry);
+            PhDereferenceObject(entry->Name);   // HACK: resolver workers may be still running and need these values
+            PhDereferenceObject(entry->TypeName);
+            PhClearReference(&entry->Target);
+            PhClearReference(&entry->TargetDrvLow);
+            PhClearReference(&entry->TargetDrvUp);
+            PhDereferenceObject(entry);
```

Comment:
Why not just let the final de-reference of the object entry clear/de-reference all the fields in the deletion routine?

Reply:
In order to do this you'll have to create an actual object type (`PhCreateObjectType`) instead of using `PhCreateAlloc`.

Reply:
Changed to
```
EtObjectEntryType = PhCreateObjectType(L"ObjectItem", 0, EtpObjectEntryDeleteProcedure);
...
entry = PhCreateObjectZero(sizeof(ET_OBJECT_ENTRY), EtObjectEntryType);
```